src/org/traccar/FilterHandler.java

```diff
@@ -84,7 +84,9 @@ public FilterHandler() {
     }
 
     private boolean filterInvalid(Position position) {
-        return filterInvalid && !position.getValid();
+        return filterInvalid && !position.getValid()
+               || position.getLatitude() > 90 || position.getLongitude() > 180
```

Comment:
You should only filter if the filter is enabled. Please be very careful and test your code before submitting anything.